src/mscorlib/shared/System/Text/ASCIIEncoding.cs

```diff
@@ -137,7 +137,7 @@ public override unsafe int GetByteCount(char* chars, int count)
 
             // Fixed doesn't like empty byte arrays
             if (bytes.Length == 0)
-                bytes = new byte[1];
+                bytes = Encoding.DummySingleByteArray;
```

Comment:
`fixed` does fine with an empty array:
https://github.com/dotnet/roslyn/issues/6707
If `bytes.Length == 0`, then `byteCount` must also be 0, and it doesn't matter what ptr we pass to GetBytes as we'll be passing in a 0-length.  So why not just remove this check altogether and just change the `&bytes[0]` below to be `bytes`?

Reply:
fixed `&bytes[0]` is an optimization on `bytes`; doesn't hurt to use the 1 array since its not used?

Reply:
> fixed &bytes[0] is an optimization on bytes; doesn't hurt to use the 1 array since its not used?

Using `bytes` just causes the compiler to emit a null check and a length check prior to doing the equivalent of `&bytes[0]`.  Switching to `bytes` puts back that null check and length check, but we get to remove the length check above it.  So all we're adding is a null check (which the JIT should be able to elide, since it would have just accessed a property on the instance, though I don't know that it does elide it), and for that we get to avoid accessing a static field.  Have you measured to see that throughput is better with the added field?

Reply:
Well using a `new byte[1]` is bad as it allocates for no particular reason; so I assume you mean the fixed change?

Didn't want to do that as switching from `&bytes[0]` to `bytes` would be reverting https://github.com/dotnet/coreclr/pull/9188

Reply:
The issue with how it is currently is I had to  [add this upstream](https://github.com/benaadams/corefx/blob/a1e792afab0bf501f03595e1a6cd2705cc053843/src/System.Runtime.Extensions/src/System/IO/StreamWriter.cs#L35-L36) to get round the allocation
```csharp
// EncoderNLS.GetBytes allocates if given an empty input array
public static readonly char[] SingleCharArray = new char[1];
```
Which means its both accessing a static field and making this `if` irrelevant (for that path).

Don't mind changing the fixed here, or using the static (as per PR), just don't think should be doing something upstream to work around it.